### PR TITLE
Add documentation for the bigquery target and edit the source docs template

### PR DESCRIPTION
**Pull request by sherifnada**
## What
1. Adds documentation for the bigquery destination
1. edits the source docs template to be generic for sources or destinations
1. Very small edits to wording on various docs

### CODE REVIEW

File: docs/integrations/destinations/bigquery.md [hunk 1]
```diff
@@ -1,2 +1,103 @@
...
+Follow the [Creating and Managing Service Account Keys](https://cloud.google.com/iam/docs/creating-managing-service-account-keys) guide to create a key. Dataline currently supports JSON Keys only, so make sure you create your key in that format. As soon as you created the key, make sure to download it, as that is the only time Google will allow you to see its contents. Once you've successfully configured BigQuery as a destination in Airbyte, delete this key from your computer. 
+
+### Setup the BigQuery destination in Airbyte
+
+You should now have all the requirements needed to configure BigQuery as a destination in the UI. You'll need the following information to configure the BigQuery destination: 
+
+* **Project ID**
+* **Dataset ID**
+* **Service Account Key**: the contents of your Service Account Key JSON file
+* **Default Target Schema:** the name of the schema where the tables will be created. In most cases, this should match the Dataset ID.
```
Comment (sherifnada) [inline docs/integrations/destinations/bigquery.md:100]: @jrhizor is this field required? the target's docs say it's not, so i'm wondering if we can just do away with it (or internally handle it by copying its value from dataset id)

File: docs/contributing/contributing-to-airbyte/updating-documentation/source-documentation-template.md [hunk 2]
```diff
@@ -1,25 +1,29 @@
-# Source Documentation Template
+---
```
Comment (michel-tricot) [inline docs/contributing/contributing-to-airbyte/updating-documentation/source-documentation-template.md:1]: John changed the capitalization of headings, make sure you adapt to it (template and all the integration docs)

File: docs/integrations/destinations/bigquery.md (hunk 1 above)
Comment (michel-tricot) [reply docs/integrations/destinations/bigquery.md:100]: I think we should do the same thing we did for filtered db and remove it completely

**Review by michel-tricot — APPROVED** [review]

File: docs/integrations/destinations/bigquery.md (hunk 1 above)
Comment (jrhizor) [reply docs/integrations/destinations/bigquery.md:100]: ^ definitely agree

**Review by jrhizor — APPROVED** [review]

**Pull request merged**